specs/DefaultDownloadDialog.md

```diff
@@ -0,0 +1,221 @@
+Default Download Dialog Layout
+===
+
+# Background
+This API gives you the ability to change the position of the default download dialog
+relative to the webview window, and to programmatically open and close the dialog.
+
+You can anchor the default download dialog to a button in your application by setting its
+position below your button, and alternating open and close calls when the button is
+clicked on.
+
+For more complex customizations, you can use the Download APIs -
+[DownloadStarting](https://docs.microsoft.com/en-us/microsoft-edge/webview2/reference/win32/icorewebview2downloadstartingeventargs?view=webview2-1.0.961.33)
+ and [DownloadOperation](https://docs.microsoft.com/en-us/microsoft-edge/webview2/reference/win32/icorewebview2downloadoperation?view=webview2-1.0.961.33) - to build custom download UI.
+
+# Examples
+```cpp
+void ViewComponent::ToggleDefaultDownloadDialog()
+{
+    if (m_webView2_6)
+    {
+        BOOL is_visible;
+        m_webView2_6->get_IsDefaultDownloadDialogVisible(&is_visible);
+        m_webView2_6->put_IsDefaultDownloadDialogVisible(!is_visible);
+    }
+}
+
+void ViewComponent::SetDefaultDownloadDialogPosition(DownloadDialogPosition target_position)
+{
+    if (m_webView2_6)
+    {
+        CHECK_FAILURE(m_webView2_6->add_DefaultDownloadDialogLayout(
+            Callback<ICoreWebView2DefaultDownloadDialogLayoutEventHandler>(
+                [this, target_position](
+                    ICoreWebView2* sender,
+                    ICoreWebView2DefaultDownloadDialogLayoutEventArgs* args) -> HRESULT {
+                    RECT bounds;
+                    CHECK_FAILURE(args->get_LocalBounds(&bounds));
+
+                    INT32 width;
+                    CHECK_FAILURE(args->get_Width(&width));
+
+                    INT32 height;
+                    CHECK_FAILURE(args->get_Height(&height));
+
+                    const int padding = 10;
+                    const int x_from_right = bounds.right - width - padding;
+                    const int y_from_bottom = bounds.bottom - height - padding;
+                    POINT new_position;
+
+                    if (target_position == DownloadDialogPosition::kTopLeft)
```

Comment:
Do we expect that end devs will generally want to set their dialogs positions like this? If so, an alternative to the event could be two properties on the settings (or wherever), one for the corner against which to position the default download dialog and another for the offset from the corner.

Reply:
The ask was to be able to control the default dialog position so that it does not cover an application's task bar, but it should keep its position relative to the webview bounds. I think the alternative would work, but I wonder about the offset:

The height of the dialog changes, so to ensure the position is within the webview bounds, the offset would have to be for the dialog corner matching the chosen corner.

But currently there are cases in which only the top-left corner of the dialog is within the webview bounds. If the dialog height or width from the default position exceed the webview bounds, the dialog is only partially in the webview. So I wonder if we should also allow offset for top-left corner to be consistent? 

cc: @bradp0721